packages/route-viewer-overlay/src/index.tsx

```diff
@@ -136,11 +137,16 @@ const RouteViewerOverlay = (props: Props): JSX.Element => {
     });
 
     if (bounds && current) {
-      util.fitMapBounds(current, bounds);
+      // Use a timeout to wait for other map rendering to finish
+      // before fitting to the route or pattern shape.
+      timeout = setTimeout(() => util.fitMapBounds(current, bounds), 500);
```

Comment:
I've noticed this happening but I'm not sure adding a delay is the best UX. Could we possibly do one `fitBounds` right away and then a second one after the delay?

Reply:
A shorter delay of 250 ms seemed to work if the concern is that someone could pan the map before the timeout expires.

Reply:
> The other alternative of course is to adjust the useEffect triggers to fire when the geometry changes and that way the entire method will fire again when the data is fully downloaded -- no timeout needed.

That is mostly the existing implementation (except that we don't compute a value or hash for the entire geometry which can be a large chunk of data).

Reply:
Do you think the hash idea is reasonable? Would prefer that to any timeouts

Reply:
With 250ms, do we still need to have an extra trigger right-away?

Reply:
I think so. 250ms is still a noticeable delay

Reply:
Changes per above added in https://github.com/opentripplanner/otp-ui/pull/507/commits/984899b77a674839bf614a0f57a73e67bb77218a.